packages/api-plugin-fulfillment-method-shipping-flat-rate/src/mutations/updateFlatRateFulfillmentRestriction.js

```diff
@@ -0,0 +1,46 @@
+import SimpleSchema from "simpl-schema";
+import ReactionError from "@reactioncommerce/reaction-error";
+import restrictionSchema from "../util/restrictionSchema.js";
+
+const inputSchema = new SimpleSchema({
+  restrictionId: String,
+  restriction: restrictionSchema,
+  shopId: String
+});
+
+
+/**
+ * @method updateFlatRateFulfillmentRestrictionMutation
+ * @summary updates a flat rate fulfillment method
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} input - Input (see SimpleSchema)
+ * @returns {Promise<Object>} An object with a `restriction` property containing the updated method
+ */
+export default async function updateFlatRateFulfillmentRestrictionMutation(context, input) {
+  const cleanedInput = inputSchema.clean(input); // add default values and such
+  inputSchema.validate(cleanedInput);
+
+  const { restriction, restrictionId, shopId } = cleanedInput;
+  const { collections } = context;
+  const { FulfillmentRestrictions } = collections;
+
+  await context.validatePermissions(`reaction:legacy:fulfillmentRestrictions:${restrictionId}`, "update", { shopId });
```

Comment:
Do permissions actually apply on the per-restriction level? That doesn't seem right

Reply:
I need to double check this. Old shipments plugins had the restrictionId level check

Reply:
@zenweasel 
Verified and confirmed that although this is present in old plugins, it does not make sense as the auth plugin would not handle it. Hence removed.